src/file_hosting/mod.rs

```diff
@@ -0,0 +1,46 @@
+use thiserror::Error;
+
+mod authorization;
+mod upload;
+mod delete;
+
+pub use authorization::authorize_account;
+pub use authorization::get_upload_url;
+pub use authorization::UploadUrlData;
+pub use authorization::AuthorizationData;
+pub use authorization::AuthorizationPermissions;
+
+pub use upload::UploadData;
+pub use upload::upload_file;
+
+pub use delete::DeleteData;
+pub use delete::delete_file_version;
+
+#[derive(Error, Debug)]
+pub enum FileHostingError {
+    #[error("Error while accessing the data from remote")]
+    RemoteWebsiteError(#[from] reqwest::Error),
+    #[error("Error while serializing or deserializing JSON")]
+    SerDeError(#[from] serde_json::Error),
+}
+
+#[cfg(test)]
```

Comment:
The tests are including too much different functions, separing them into multiple ones and one final that test their integration together is needed

Reply:
the problem with that is all of the methods depend on each other so doing that is basically impossible